api/trace/testtrace/span_test.go

```diff
@@ -120,6 +121,96 @@ func TestSpan(t *testing.T) {
 		})
 	})
 
+	t.Run("#RecordError", func(t *testing.T) {
+		t.Run("records an error", func(t *testing.T) {
+			t.Parallel()
+
+			e := matchers.NewExpecter(t)
+
+			tracer := testtrace.NewTracer()
+			ctx, span := tracer.Start(context.Background(), "test")
+
+			subject, ok := span.(*testtrace.Span)
+			e.Expect(ok).ToBeTrue()
+
+			errMsg := "test error message"
+			testTime := time.Now()
+			subject.RecordError(ctx, errors.New(errMsg), trace.WithErrorTime(testTime))
```

Comment:
Could we use the testError thing here too? I wonder if you could export it in some internal package and use it here to avoid code duplication.